esmvaltool/diag_scripts/validation.py

```diff
@@ -0,0 +1,306 @@
+    logger.debug("Time-averaged experiment %s", exper_cube)
+    if obs_list:
+        obs_cube_list = []
+        for obs in obs_list:
+            obs_file = obs['filename']
+            logger.info("Loading %s", obs_file)
+            obs_cube = iris.load_cube(obs_file)
+            obs_cube = obs_cube.collapsed('time', iris.analysis.MEAN)
+            logger.debug("Time-averaged obs %s", obs_cube)
+            obs_cube_list.append(obs_cube)
+    else:
+        obs_cube_list = None
+
+    return ctrl_cube, exper_cube, obs_cube_list
+
+
+def apply_seasons(data_set_dict):
+    """Extract seaons and apply a time mean per season"""
+    data_file = data_set_dict['filename']
+    logger.info("Loading %s for seasonal extraction", data_file)
+    data_cube = iris.load_cube(data_file)
+    seasons = ['DJF', 'MAM', 'JJA', 'SON']
+    season_cubes = [extract_season(data_cube, season) for season in seasons]
+    season_meaned_cubes = [
+        season_cube.collapsed('time', iris.analysis.MEAN)
+        for season_cube in season_cubes
+    ]
+
+    return season_meaned_cubes
+
+
+def coordinate_collapse(data_set, cfg):
+    """Perform coordinate-specific collapse and (if) area slicing and mask"""
+    # see what analysis needs performing
+    analysis_type = cfg['analysis_type']
+
+    # if subset on LAT-LON
+    if 'lat_lon_slice' in cfg:
+        start_longitude = cfg['lat_lon_slice']['start_longitude']
+        end_longitude = cfg['lat_lon_slice']['end_longitude']
+        start_latitude = cfg['lat_lon_slice']['start_latitude']
+        end_latitude = cfg['lat_lon_slice']['end_latitude']
+        data_set = area_slice(data_set, start_longitude, end_longitude,
+                              start_latitude, end_latitude)
+
+    # if apply mask
+    if '2d_mask' in cfg:
+        mask_file = os.path.join(cfg['2d_mask'])
+        mask_cube = iris.load_cube(mask_file)
+        if 'mask_threshold' in cfg:
+            thr = cfg['mask_threshold']
+            data_set.data = np.ma.masked_array(
+                data_set.data, mask=(mask_cube.data > thr))
+        else:
+            logger.warning('Could not find masking threshold')
+            logger.warning('Please specify it if needed')
+            logger.warning('Masking on 0-values = True (masked value)')
+            data_set.data = np.ma.masked_array(
+                data_set.data, mask=(mask_cube.data == 0))
+
+    # if zonal mean on LON
+    if analysis_type == 'zonal_mean_longitude':
+        data_set = data_set.collapsed('longitude', iris.analysis.MEAN)
+
+    # if zonal mean on LAT
+    if analysis_type == 'zonal_mean_latitude':
+        data_set = data_set.collapsed('latitude', iris.analysis.MEAN)
+
+    # if vertical mean
+    elif analysis_type == 'vertical_mean':
+        data_set = data_set.collapsed('pressure', iris.analysis.MEAN)
+
+    return data_set
+
+
+def do_preamble(cfg):
+    """Execute some preamble functionality"""
+    # prepare output dirs
```

Comment:
There is no need to create those output dirs, they are already created by the function `run_diagnostic`.

Reply:
removed, man, cheers